gitlab/resource_gitlab_branch_protection.go

```diff
@@ -0,0 +1,128 @@
+package gitlab
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabBranchProtection() *schema.Resource {
+	acceptedAccessLevels := []string{
```

Comment:
How could I set access level to "No one"?

Reply:
is it possible to get that list from elsewhere?

Reply:
Could you please add `no one` and `maintainer` here?

Reply:
> is it possible to get that list from elsewhere?

@roidelapluie should we sync the branch with the upstream master and then use the accessLevel map from https://github.com/terraform-providers/terraform-provider-gitlab/blob/master/gitlab/util.go#L140